ts/torch_handler/base_handler.py

```diff
@@ -78,38 +81,101 @@ def initialize(self, context):
         self.manifest = context.manifest
 
         model_dir = properties.get("model_dir")
-        model_pt_path = None
+        self.model_pt_path = None
         if "serializedFile" in self.manifest["model"]:
             serialized_file = self.manifest["model"]["serializedFile"]
-            model_pt_path = os.path.join(model_dir, serialized_file)
+            self.model_pt_path = os.path.join(model_dir, serialized_file)
+
+        if self.model_pt_path.endswith("onnx"):
+            try:
+                # import numpy as np
+                import onnxruntime as ort
+
+                onnx_enabled = True
+                logger.info("ONNX enabled")
+            except ImportError as error:
+                onnx_enabled = False
+                logger.warning("proceeding without onnxruntime")
 
         # model def file
         model_file = self.manifest["model"].get("modelFile", "")
 
         if model_file:
             logger.debug("Loading eager model")
-            self.model = self._load_pickled_model(model_dir, model_file, model_pt_path)
+            self.model = self._load_pickled_model(
+                model_dir, model_file, self.model_pt_path
+            )
             self.model.to(self.device)
-        else:
-            logger.debug("Loading torchscript model")
-            if not os.path.isfile(model_pt_path):
-                raise RuntimeError("Missing the model.pt file")
+            self.model.eval()
+
+        # Convert your model by following instructions: https://pytorch.org/tutorials/intermediate/nvfuser_intro_tutorial.html
+        # For TensorRT support follow instructions here: https://pytorch.org/TensorRT/getting_started/getting_started_with_python_api.html#getting-started-with-python-api
+        elif self.model_pt_path.endswith(".pt"):
+            self.model = self._load_torchscript_model(self.model_pt_path)
+            self.model.eval()
+
+        # Convert your model by following instructions: https://pytorch.org/tutorials/advanced/super_resolution_with_onnxruntime.html
+        elif self.model_pt_path.endswith(".onnx") and onnx_enabled:
+            # self.model = self._load_onnx_model(self.model_pt_path)
+            providers = (
+                ["CUDAExecutionProvider", "CPUExecutionProvider"]
```

Comment:
I wonder if it should be "CUDAExecutionProvider" instead of "  ["CUDAExecutionProvider", "CPUExecutionProvider"]"

Reply:
No it's both because you may need to fallback to CPU for some operations, the tutorial I link in tests shows this very pattern